src/mainviews/list-jobs.tsx

```diff
@@ -185,7 +220,29 @@ function ListJobDefinitionsTable(props: ListJobDefinitionsTableProps) {
   const [deletedRows, setDeletedRows] = useState<
     Set<Scheduler.IDescribeJobDefinition['job_definition_id']>
   >(new Set());
-  const [displayError, setDisplayError] = useState<string | null>(null);
+  // Alerts to display with varying severities
+  const [displayError, setDisplayError] = useState<React.ReactNode | null>(
+    null
+  );
+
+  // Display creation message
+  const now = new Date().getTime();
+
+  const successMessage =
+    props.newlyCreatedId !== undefined &&
+    props.newlyCreatedName !== undefined &&
+    props.successMessageExpiration !== undefined &&
+    now <= props.successMessageExpiration
+      ? trans.__(
+          'Your job definition "%1" has been created. ' +
+            'If you do not see it in the list below, please reload the list in a few seconds.',
+          props.newlyCreatedName
+        )
+      : null;
+
+  const [displayInfo, setDisplayInfo] = useState<React.ReactNode | null>(
+    successMessage
+  );
```

Comment:
1. This state management logic is faulty. It only clears the error on re-render, not every minute as expected.

2. The success message expiration time is passed as a prop even though it's a constant value. This makes it confusing because now you've split the expiration logic into `notebook-jobs-panel.tsx` for no reason. Instead, hardcode a constant like `const INFO_BANNER_EXPIRATION = 60`, expose it here in the list view, and remove `successMessageExpiration`.

3. You are duplicating logic in both the ListJobs and ListJobDefinitions table, when you could instead place this logic inside their shared parent component `<NotebookJobsList />`

In short, your component logic should look something like this:

```tsx
// outside component at file-scope
const INFO_BANNER_EXPIRATION = 60

// inside <NotebookJobsList /> component
const [displayInfo, setDisplayInfo] = useState<React.ReactNode>(null);
useEffect(() => {
  if (!props.newlyCreatedId && !props.newCreatedName) return;
  const successMessage = ...;
  setDisplayInfo(successMessage)
  setTimeout(() => setDisplayInfo(null), INFO_BANNER_EXPIRATION);
}, [props.newlyCreatedId, props.newlyCreatedName]);
```

Reply:
The reason for passing the expiration time as a prop is so that it gets serialized, not so that the banner disappears after 60 seconds (which is not a requirement). This solves the problem of:

1. User creates a job
2. User is brought to the list jobs page which displays a success banner
3. User closes their browser
4. Hours later, the user reconnects to JupyterLab
5. User should see the list jobs page without a success banner (because they haven't done anything recently).

Reply:
OK, then it sounds like the real solution is to not have the newly created ID and name serialized in the model, correct? IOW, just have them be instance properties on the NotebookJobsPanel class.

Reply:
I don't see a use case for making our logic more complex to support the use case of a persistent banner across reloads within 60 seconds.

Reply:
Deduplicated status message. Removed the 60-second timeout.